Remove WidgetWithScript - potentially unsafe and no longer used

lb- opened the pull request:
- Since the work on adopting `render_js_init` for the `BaseChooser`, we no longer need WidgetWithScript - #8501 (could not find the exact PR, but this was prep work for it).
- Note: This is not an active risk but is an approach we should actively try to avoid, additionally we now have a solid approach to avoid this kind of thing (init scripts) with Stimulus.
- With the recent security alerts I thought it would be good to remove this unused widget, except for the bare minimum needed for some of the test app items.
- We probably need to add something to the release notes as maybe other packages are using this.

nmorduch commented:
Can the tests not work with a widget pattern we do want to recommend people use? It feels a bit weird to test with something we are actively saying we don't want to use anymore.

lb- commented:
@nmorduch  good call - I guess we should not be writing tests for practices we want to move away from and not recommend other developers implement.

I have reworked this clean up to avoid the inline script approach completely and kept it simple with testing what we care about (that different HTML based on the widget's ID is rendered).

Rebased & pushing changes now.

gasman commented:
Definitely worthwhile, yep! I remember WidgetWithScript being a particular bugbear when refactoring the chooser widgets, and possibly something standing in the way of further cleanups to the always-on Javascript too - so if we're at a point where it's feasible to drop it, that's very good news.

lb- commented:
Update: I am still intending to resolve this PR, just waiting for #10590 to be resolved. This way I can add a section to that new part of the docs on how to build similar behaviour with Stimulus/Django media for our release notes.

lb- commented:
PR updated off the back of the Stimulus client-side docs - ready for a fresh review.

No longer removing `WidgetWithScript` but instead setting up for a deprecation path with a documented recommendation for these kinds of widgets that pull in data from Python into JS that can appear Dynamically.

@gasman

lb- commented:
Bumping to 5.3 - will update the PR accordingly in the next little while.

lb- commented:
OK - good to go again on the 5.3 code, added a skeleton 5.3 release notes page as an initial commit on this branch also.

lb- commented:
Just rebased.

gasman reviewed (approved):
Thanks @lb-! All looks good to me (except that the deprecation warning will now be RemovedInWagtail70Warning, which I'll update when I merge).

gasman commented:
Merged in e1693ee5db1f34d6c31fd3f643b53365b44c2087.

lb- commented:
Thank you
